CMakeLists.txt

```diff
@@ -1,31 +1,81 @@
-cmake_minimum_required(VERSION 3.12)
+cmake_minimum_required(VERSION 3.14)
 
 include(pico_sdk_import.cmake)
 
 project(picoprobe)
 
+# If the env var 'PICO_BOARD' is set in the calling shell
+# eg. export PICO_BOARD=adafruit_qtpy_rp2040
+# then use it here
+if (NOT DEFINED ENV{PICO_BOARD})
+    # Otherwise use PICO_BOARD as defined here, which is handy when
+    # you're running CMake from an IDE (eg. VSCode) rather than the
+    # commmand line.
+    # Set 'PICO_BOARD' to a board name as defined in
+    # https://github.com/raspberrypi/pico-sdk/tree/master/src/boards/include/boards
+    #
+    # For example,
+    #    "adafruit_qtpy_rp2040"    for the Adafruit QTPy
+    #    "adafruit_feather_rp2040" for the Adafruit Feather RP2040
+    #    "sparkfun_promicro"       for the SparkFun Pro Micro RP2040
+    #
+    # The default is "pico"
+
+    # Set PICO_BORD here so that the SDK correctly applies it
+    set(PICO_BOARD "pico")
+    message("Environment variable PICO_BOARD undefined -- using \"pico\"")
+else()
+    # Set PICO_BORD here, from the env var, so that the SDK correctly applies it
+    set(PICO_BOARD "$ENV{PICO_BOARD}")
+    message("Environment variable PICO_BOARD defined -- using $ENV{PICO_BOARD}")
+endif()
+
+set(EXECUTABLE_NAME "picoprobe_${PICO_BOARD}")
+
 pico_sdk_init()
 
-add_executable(picoprobe
-        src/led.c
-        src/main.c
-        src/usb_descriptors.c
-        src/probe.c
-        src/cdc_uart.c
-        src/get_serial.c
+add_executable(${EXECUTABLE_NAME}
+               src/led.c
+               src/main.c
+               src/usb_descriptors.c
+               src/probe.c
+               src/cdc_uart.c
+               src/get_serial.c
 )
 
+if (${PICO_BOARD} STREQUAL "adafruit_qtpy_rp2040")
+    message("Building for Adafruit QTPy RP2040 with WS2812")
+elseif (${PICO_BOARD} STREQUAL "adafruit_feather_rp2040")
+    message("Building for Adafruit Feather RP2040 with WS2812")
+elseif (${PICO_BOARD} STREQUAL "sparkfun_promicro")
+    message("Building for SparkFun Pro Micro RP2040 with WS2812")
+else()
+    message("Building for Raspberry Pi Pico")
+endif()
+
+if (NOT ${PICO_BOARD} STREQUAL "pico")
+    target_sources(${EXECUTABLE_NAME} PRIVATE src/ws2812.c)
```

Comment:
This seems to be assuming that any board that *isn't* a Raspberry Pi Pico _does_ have a WS2812 LED, which doesn't feel like a very flexible / easily-extendable solution? :man_shrugging: